src/base_media_base.h

```diff
@@ -279,8 +274,8 @@ static const uint NUM_SONGS_AVAILABLE = 1 + NUM_SONG_CLASSES * NUM_SONGS_CLASS;
 static const uint NUM_SONGS_PLAYLIST  = 32;
 
 /* Functions to read DOS music CAT files, similar to but not quite the same as sound effect CAT files */
-char *GetMusicCatEntryName(const char *filename, size_t entrynum);
-byte *GetMusicCatEntryData(const char *filename, size_t entrynum, size_t &entrylen);
+char *GetMusicCatEntryName(const std::string &filename, size_t entrynum);
```

Comment:
To be functionally the same, it should return `std::optional<std::string>` as with just `std::string` an empty name will cause the 'song is missing' warning, where it did not do that before.
The main problem/issue with these PRs is that they quickly rabbit hole out of control if you keep unravelling the thread. Here it is returning a completely new string that does not have anything in common with the input, so that seems like a good place to stop the unravelling.